packages/itwin/grouping-mapping-widget/src/test/QueryBuilder.testdata.json

```diff
@@ -0,0 +1,808 @@
+                  "isPolymorphicTargetClass": false,
+                  "relationshipInfo": {
+                    "id": "id5",
+                    "name": "A:E",
+                    "label": "AE label"
+                  },
+                  "isPolymorphicRelationship": true,
+                  "isForwardRelationship": true
+                }
+              ],
+              "contentClassInfo": {
+                "name": "A:B"
+              }
+            }
+          }
+        }
+      ]
+    },
+    {
+      "name": "when no properties added, return empty string",
+      "expectedResult": "",
+      "operations": []
+    },
+    {
+      "name": "when property is float, return query string with ROUND",
+      "expectedResult": "SELECT A.B.ECInstanceId FROM A.B WHERE ROUND(A.B.somePropName, 4)=3.1416",
+      "operations": [
+        {
+          "expectedResult": true,
+          "operationType": "addProperty",
+          "propertyRecord": {
+            "value": {
+              "valueFormat": 0,
+              "value": 3.14159
+            },
+            "property": {
+              "typename": "notNavigation"
+            }
+          },
+          "propertiesField": {
+            "properties": [
+              {
+                "property": {
+                  "name": "somePropName",
+                  "classInfo": {
+                    "name": "A:B"
+                  }
+                }
+              }
+            ],
+            "type": {
+              "typeName": "number"
+            },
+            "parent": {
+              "pathToPrimaryClass": []
+            }
+          }
+        }
+      ]
+    },
+    {
+      "name": "when property needsQuotes, return query string with property value in quotes",
+      "expectedResult": "SELECT A.B.ECInstanceId FROM A.B WHERE A.B.somePropName=0",
+      "operations": [
+        {
+          "expectedResult": true,
+          "operationType": "addProperty",
+          "propertyRecord": {
+            "value": {
+              "valueFormat": 0,
+              "value": 0
+            },
+            "property": {
+              "typename": "notNavigation"
+            }
+          },
+          "propertiesField": {
+            "properties": [
+              {
+                "property": {
+                  "name": "somePropName",
+                  "classInfo": {
+                    "name": "A:B"
+                  }
+                }
+              }
+            ],
+            "type": {
+              "typeName": "number"
+            },
+            "parent": {
+              "pathToPrimaryClass": []
+            }
+          }
+        }
+      ]
+    },
+    {
+      "name": "when property is category, return a category query string",
+      "expectedResult": "SELECT BisCore.B.ECInstanceId FROM BisCore.B JOIN bis.Category ON bis.Category.ECInstanceId = bis.GeometricElement3d.category.id AND ((bis.Category.CodeValue='') OR (bis.Category.UserLabel=''))",
```

Comment:
It would be good to see `Category.CodeValue` and `Category.UserLabel` values compared to non empty strings.

Reply:
Fixed

Reply:
looks good